test/helpers/add_retries_test_helper.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# Usage: Extend a descendant of Minitest::Test with this module
+# and then add "__Nretries" to the end of the names of the tests you want to retry up to N times if they fail
+module AddRetriesTestHelper
+  # ENV["VERBOSE"] example:
+  #  KubernetesDeployTest#test_hpa_can_be_successful_and_gets_pruned__2retries 0.58 = F
+  #  KubernetesDeployTest#test_hpa_can_be_successful_and_gets_pruned__2retries (Retry 1/2) 0.53 = F
+  #  KubernetesDeployTest#test_hpa_can_be_successful_and_gets_pruned__2retries (Retry 2/2) 0.52 = F
+
+  # !ENV["VERBOSE"] example:
+  #   F Retry1/2:F Retry2/2:F
+  def run_one_method(klass, method_name, reporter)
```

Comment:
Although a patch like this seems to be the way all the retry-related extensions for minitest are written (it is apparently factored separately from `Minitest.run_one_method` to allow this), doing so is fundamentally incompatible with running tests in parallel. Our `super` here is `Minitest::Parallel`, which has its own special version that just pushes tests to a queue--they aren't meant to actually run right here. That explains why all my CI runs were slowish even though HPA tests only failed once across all of them (the HPA tests were blocking). I would have to mess with `Minitest.run_one_method` instead, which is REALLY sketchy and doesn't have `reporter` access. 👎